app/uk/gov/hmrc/eventhub/controllers/PublishController.scala

```diff
@@ -38,10 +38,11 @@ class PublishController @Inject() (cc: ControllerComponents, publisher: Publishe
             case Right(_) => Created
             case Left(error) =>
               error match {
-                case e: DuplicateEvent        => Created(e.message)
-                case e: NoEventTopic          => NotFound(e.message)
-                case e: NoSubscribersForTopic => Created(e.message)
-                case e                        => InternalServerError(e.message)
+                case e: NoMatchingConfigurationPath => Created(e.message)
+                case e: DuplicateEvent              => Created(e.message)
+                case e: NoEventTopic                => NotFound(e.message)
+                case e: NoSubscribersForTopic       => Created(e.message)
+                case e                              => InternalServerError(e.message)
```

Comment:
this is not so much about this PR, but I'm just wondering, if we could re-think the name of those? when it's something not configured, maybe we could say "XNotConfigured", and when it's something missing from the message we could say something like "XMissing"? not sure